nixos/modules/system/boot/plymouth.nix

```diff
@@ -160,12 +160,22 @@ in
     systemd.services.plymouth-poweroff.wantedBy = [ "poweroff.target" ];
     systemd.services.plymouth-reboot.wantedBy = [ "reboot.target" ];
     systemd.services.plymouth-read-write.wantedBy = [ "sysinit.target" ];
+    systemd.services.plymouth-switch-root-initramfs.wantedBy = [
+      "halt.target"
+      "kexec.target"
+      "poweroff.target"
+      "reboot.target"
+      "generate-shutdown-ramfs.service"
+      "plymouth-switch-root-initramfs.service"
+    ];
     systemd.services.systemd-ask-password-plymouth.wantedBy = [ "multi-user.target" ];
```

Comment:
It's not obvious to me why these are added in systemd stage 2 and not in stage 1. It's in stage 1 that we'll ask the user for a password to decrypt disks.